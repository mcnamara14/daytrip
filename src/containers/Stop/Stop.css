.plan {
  width: 100%;
  display: flex;
  justify-content: space-between;
  align-items: center;
  box-shadow: 1px 1px 1px #ccc;
  border: 1px solid #f8f8f8;
}

.planContainer {
  padding-bottom: 65px;
  border-bottom: 1px solid #eee;
}

.plan h2 {
  margin-top: 20px;
  color: #444;
  text-transform: uppercase;
  font-size: 1.1rem;
  margin-bottom: 10px;
}

.plan article {
  width: 29%;
  text-align: left;
  padding: 30px;
}

.stars {
  float: left;
  margin: 0 10px 0 0;
  width: 100px;
  position: relative;
}

.redStars{
  border: 0;
  left: 0;
  overflow: hidden;
  position: absolute;
  top: 0;
 } 

 .stars img {
   height: auto;
   max-width: 100px;
 }

.reviewCount {
  color: #888;
  font-size: 0.9rem;
}

.price {
  font-size: 1.2rem;
  color: #666;
}

.location {
  color: #888;
  font-size: 0.85rem;
  margin: 0 0 25px 0;
}

.stopImage {
  width: 100%;
  height: 190px;
  border-radius: 5px 5px 0 0;
  display: block;
  background-size: cover;
  background-position: center;
  border: 1px solid #eee;
}

.stopInfo {
  height: 185px;
  padding: 0 25px;
}

.routeImg {
  margin-top: 0;
  width: 50px;
}

.reviewContainer {
  height: 19px;
  display: flex;
  align-items: center;
}