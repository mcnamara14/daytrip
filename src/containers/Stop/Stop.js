import React, { Component } from 'react';
import './Stop.css';
import PropTypes from 'prop-types';

export class Stop extends Component {

  render() {
    const { 
      title, 
      rating, 
      location, 
      price, 
      reviews, 
      image, 
      index, 
      type 
    } = this.props;
    const backgroundImage = {backgroundImage: "url(" + image + ")"};
    const className = `${type}-${index} ${type}Stop`;
    const review = rating * 20;

    return (
      <article className={className}  >
        <div className="stopImage" style={backgroundImage}></div>
        <div className="stopInfo">
          <div className="stopHeader">
            <h2>{title}</h2>
            { type === 'restaurant' || type === 'bar' ?
              <div className="reviewContainer">
                <div className="stars">
                  <div>
                    <img src={require('./assets/stars-gray.png')} alt="Stars gray"/>
                  </div>
                  <div className="redStars" style={{ "width": `${review}%` }}>
                    <img src={require('./assets/stars-red.png')} alt="Stars red"/>
                  </div>
                </div>
                <p className="reviewCount">{reviews} reviews</p>
              </div> :
              null
            }
          </div>
          { type === 'event' ? 
            <div className="stopInfo">
              <div className="location">
                <img src={require('./assets/location-marker.png')} alt="Location marker icon"/>
                {location} 
              </div>
              <div className="stopPrice">
                <img src={require('./assets/price-tag.png')} alt="Price tag icon"/>
                {price} 
              </div>
            </div>  :
            <div className="stopInfo">
              <div className="location">
                <img src={require('./assets/location-marker.png')} alt="Location marker icon"/>
                {location} 
              </div>
              <div className="stopPrice">
                <img src={require('./assets/price-tag.png')} alt="Price tag icon"/>
                {price} 
              </div>
            </div> 
          }
        </div>
      </article>
    );
  }
}

Stop.propTypes = {
  title: PropTypes.string,
  rating: PropTypes.string,
  location: PropTypes.string,
  price: PropTypes.string,
  reviews: PropTypes.string,
  image: PropTypes.string,
  index: PropTypes.string,
  type: PropTypes.string
};

export default Stop;