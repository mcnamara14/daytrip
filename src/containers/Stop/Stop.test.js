import React from 'react';
import ReactDOM from 'react-dom';
import { Stop } from './Stop';
import { shallow } from 'enzyme';

it('should match the snapshot', () => {
  const wrapper = shallow(<Stop />);

  expect(wrapper).toMatchSnapshot();
});
