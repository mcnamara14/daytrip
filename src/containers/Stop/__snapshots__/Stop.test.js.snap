// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stop should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Stop
    type="bar"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="stopImage"
          style={
                    Object {
                              "backgroundImage": "url(undefined)",
                            }
          }
/>,
        <div
          className="stopInfo"
>
          <h2 />
          <p
                    className="location"
          />
          <div
                    className="reviewContainer"
          >
                    <div
                              className="stars"
                    >
                              <div>
                                        <img
                                                  src="stars-gray.png"
                                        />
                              </div>
                              <div
                                        className="redStars"
                                        style={
                                                  Object {
                                                            "width": "NaN%",
                                                          }
                                        }
                              >
                                        <img
                                                  src="stars-red.png"
                                        />
                              </div>
                    </div>
                    <p
                              className="reviewCount"
                    >
                               reviews
                    </p>
          </div>
          <p
                    className="price"
          />
</div>,
      ],
      "className": "bar-undefined",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "stopImage",
          "style": Object {
            "backgroundImage": "url(undefined)",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2 />,
            <p
              className="location"
/>,
            <div
              className="reviewContainer"
>
              <div
                            className="stars"
              >
                            <div>
                                          <img
                                                        src="stars-gray.png"
                                          />
                            </div>
                            <div
                                          className="redStars"
                                          style={
                                                        Object {
                                                                      "width": "NaN%",
                                                                    }
                                          }
                            >
                                          <img
                                                        src="stars-red.png"
                                          />
                            </div>
              </div>
              <p
                            className="reviewCount"
              >
                             reviews
              </p>
</div>,
            <p
              className="price"
/>,
          ],
          "className": "stopInfo",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "location",
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="stars"
>
                  <div>
                                    <img
                                                      src="stars-gray.png"
                                    />
                  </div>
                  <div
                                    className="redStars"
                                    style={
                                                      Object {
                                                                        "width": "NaN%",
                                                                      }
                                    }
                  >
                                    <img
                                                      src="stars-red.png"
                                    />
                  </div>
</div>,
                <p
                  className="reviewCount"
>
                   reviews
</p>,
              ],
              "className": "reviewContainer",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div>
                      <img
                                            src="stars-gray.png"
                      />
</div>,
                    <div
                      className="redStars"
                      style={
                                            Object {
                                                                  "width": "NaN%",
                                                                }
                      }
>
                      <img
                                            src="stars-red.png"
                      />
</div>,
                  ],
                  "className": "stars",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <img
                        src="stars-gray.png"
/>,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "src": "stars-gray.png",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "img",
                    },
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <img
                        src="stars-red.png"
/>,
                      "className": "redStars",
                      "style": Object {
                        "width": "NaN%",
                      },
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "src": "stars-red.png",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "img",
                    },
                    "type": "div",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    undefined,
                    " reviews",
                  ],
                  "className": "reviewCount",
                },
                "ref": null,
                "rendered": Array [
                  undefined,
                  " reviews",
                ],
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "price",
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="stopImage"
            style={
                        Object {
                                    "backgroundImage": "url(undefined)",
                                  }
            }
/>,
          <div
            className="stopInfo"
>
            <h2 />
            <p
                        className="location"
            />
            <div
                        className="reviewContainer"
            >
                        <div
                                    className="stars"
                        >
                                    <div>
                                                <img
                                                            src="stars-gray.png"
                                                />
                                    </div>
                                    <div
                                                className="redStars"
                                                style={
                                                            Object {
                                                                        "width": "NaN%",
                                                                      }
                                                }
                                    >
                                                <img
                                                            src="stars-red.png"
                                                />
                                    </div>
                        </div>
                        <p
                                    className="reviewCount"
                        >
                                     reviews
                        </p>
            </div>
            <p
                        className="price"
            />
</div>,
        ],
        "className": "bar-undefined",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "stopImage",
            "style": Object {
              "backgroundImage": "url(undefined)",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2 />,
              <p
                className="location"
/>,
              <div
                className="reviewContainer"
>
                <div
                                className="stars"
                >
                                <div>
                                                <img
                                                                src="stars-gray.png"
                                                />
                                </div>
                                <div
                                                className="redStars"
                                                style={
                                                                Object {
                                                                                "width": "NaN%",
                                                                              }
                                                }
                                >
                                                <img
                                                                src="stars-red.png"
                                                />
                                </div>
                </div>
                <p
                                className="reviewCount"
                >
                                 reviews
                </p>
</div>,
              <p
                className="price"
/>,
            ],
            "className": "stopInfo",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "location",
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="stars"
>
                    <div>
                                        <img
                                                            src="stars-gray.png"
                                        />
                    </div>
                    <div
                                        className="redStars"
                                        style={
                                                            Object {
                                                                                "width": "NaN%",
                                                                              }
                                        }
                    >
                                        <img
                                                            src="stars-red.png"
                                        />
                    </div>
</div>,
                  <p
                    className="reviewCount"
>
                     reviews
</p>,
                ],
                "className": "reviewContainer",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <div>
                        <img
                                                src="stars-gray.png"
                        />
</div>,
                      <div
                        className="redStars"
                        style={
                                                Object {
                                                                        "width": "NaN%",
                                                                      }
                        }
>
                        <img
                                                src="stars-red.png"
                        />
</div>,
                    ],
                    "className": "stars",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <img
                          src="stars-gray.png"
/>,
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "src": "stars-gray.png",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "img",
                      },
                      "type": "div",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <img
                          src="stars-red.png"
/>,
                        "className": "redStars",
                        "style": Object {
                          "width": "NaN%",
                        },
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "src": "stars-red.png",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "img",
                      },
                      "type": "div",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      undefined,
                      " reviews",
                    ],
                    "className": "reviewCount",
                  },
                  "ref": null,
                  "rendered": Array [
                    undefined,
                    " reviews",
                  ],
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "price",
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
